functions/src/winloss.ts

```diff
@@ -0,0 +1,68 @@
+import { firestore } from "firebase-admin";
+import * as functions from "firebase-functions";
+import { getPrivatePlayerCollection, lobbyCollection } from "./firestore-collections";
+import { isLobbyRequest } from "./firestore-functions-types";
+import { Lobby } from "./firestore-types/lobby";
+
+export const lobbyReturn = functions.https.onCall(async (data: unknown) => {
+  // validate code
+  if (!isLobbyRequest(data)) {
+    return { error: "Invalid lobby code!" };
+  }
+  // get lobby doc
+  const lobby = await lobbyCollection.doc(data.code).get();
+  if (lobby.exists === false) {
+    return { error: "Lobby doesn't exist!" };
+  }
+  return lobbyCollection.doc(data.code).update({ state: "WAIT" });
+});
+
+async function getPrivatePlayer(lobbyDocRef: firestore.DocumentReference<Lobby>, id: string) {
+  const privatePlayerCollection = getPrivatePlayerCollection(lobbyDocRef);
+
+  const privatePlayerDoc = privatePlayerCollection.doc(id);
+  const privatePlayer = await privatePlayerDoc.get();
+  return privatePlayer.data();
+}
+
+export const getLobbyRole = functions.https.onCall(async (data: unknown, context) => {
```

Comment:
this should probobly not be run for every client. it should run only once and then the results should be put in the database so everyone can see it